### PR TITLE
Assignment4.

### PR DESCRIPTION
**Pull request by CalumMackenzie-EWU**
I've managed to get the Implementation of the CRUD operations for the User controller. 
No tests implemented at all.

*edit: Rest file now working.
*edit2: changed the title since I dont think Ill be able to get the testing ready in time.

### CODE REVIEW

**Comment by Thunderbird145** [comment]
I'll Review This.

**Comment by Thunderbird145** [comment]
Create a UserController in the API projects
Implement endpoints for:
Retrieving all users ✔
Retriving a single user by id ✔
Creating a user ✔
Updating a user ✔
Deleting a user ✔
The UserController should function. ✔
Include a .rest file for interacting with the controller (REST Client). ✔
The IUserRepository interface is provided for you. You will need to provide an implementation of this interface within the SecretSanta.Business project. This implementation can simply store the user within a list, dictionary, array, or similar. ✔
All test projects should be within a test directory ❌
Create a unit test project for the SecretSanta.Business project. ❌
Unit test all public methods ❌
Ensure the project is named with a ".Tests" suffix ❌
Ensure test class is named appropriately for the class being tested: ❌
Name all tests in the form <MethodBeingTested>_<ConditionBeingTested>_<ExpectedOutcome>(). You can add <NoWarn>CA1707;INTL0003</NoWarn> to remove the naming warnings. ❌
Create a Unit Test project for the SecretSanta.Api project. ❌
Test all controller public methods. It is up to you to determine the best way to provide the controller's dependencies. ❌
Ensure the project is named with a ".Tests" suffix ❌

**Comment by Thunderbird145** [comment]
About halfway done, probably more tests than necessary in the .rest file, but otherwise, good job so far!

File: SecretSanta/src/SecretSanta.Api/Dto/UpdateUser.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+namespace SecretSanta.Api.Dto
+{
+    public class UpdateUser{
+        public string? FirstName { get; set; }
+        public string? LastName { get; set; }
+        public int Id{get;set;}
```
Comment (Keboo) [inline SecretSanta/src/SecretSanta.Api/Dto/UpdateUser.cs:6]: For the update case the ID is passed in separately. If your controller, this property is (appropriately) ignored.
```suggestion
```

**Review by Keboo — COMMENTED** [review]
What is here looks good. I would review the Assignment4Complete branch and let me know if you have any questions on the testing work that was implemented there.

- Create a UserController in the API projects
  - Implement endpoints for:
    - Retrieving all users ✔
    - Retriving a single user by id ✔
    - Creating a user ✔
    - Updating a user ✔
    - Deleting a user ✔
  - The UserController should function. ✔
  - Include a `.rest` file for interacting with the controller. ✔
- The `IUserRepository` interface is provided for you. You will need to provide an implementation of this interface within the SecretSanta.Business project. This implementation can simply store the user within a list, dictionary, array, or similar. ✔
- All test projects should be within a test directory ❌
- Create a unit test project for the SecretSanta.Business project. ❌
  - Unit test all public methods ❌
  - Ensure the project is named with a ".Tests" suffix ❌
  - Ensure test class is named appropriately for the class being tested: ❌
  - Name all tests in the form `<MethodBeingTested>_<ConditionBeingTested>_<ExpectedOutcome>()`. You can add  `<NoWarn>CA1707;INTL0003</NoWarn>` to remove the naming warnings. ❌
- Create a Unit Test project for the SecretSanta.Api project. ❌
  - Test all controller public methods. It is up to you to determine the best way to provide the controller's dependencies. ❌
  - Ensure the project is named with a ".Tests" suffix ❌

**Pull request closed without merging**